src/target/renesas.c

```diff
@@ -91,6 +91,102 @@ typedef enum {
+	{PNR_SERIES_RA4M1, 0x01010008, 44, 0, 0, 0x40100000, 0x2000, 0, 0, 0, 0, 0x20000000, 0x8000, 0, 0, 0, 0,
+		RENESAS_FLASH_MF3, RENESAS_LOCATION_FRT},
+	{PNR_SERIES_RA4M2, 0x0100a100, 512, 0, 0, 0x08000000, 0x2000, 0x010080f0, 196, 0, 0, 0x20000000, 0x20000, 0, 0,
+		0x28000000, 0x400, RENESAS_FLASH_RV40, RENESAS_LOCATION_FIXED2},
+	{PNR_SERIES_RA4M3, 0x0100a100, 512, 0, 0, 0x08000000, 0x2000, 0x010080f0, 196, 0, 0, 0x20000000, 0x20000, 0, 0,
+		0x28000000, 0x400, RENESAS_FLASH_RV40, RENESAS_LOCATION_FIXED2},
+	{PNR_SERIES_RA4E1, 0x0100a100, 512, 0, 0, 0x08000000, 0x2000, 0x010080f0, 196, 0, 0, 0x20000000, 0x20000, 0, 0,
+		0x28000000, 0x400, RENESAS_FLASH_RV40, RENESAS_LOCATION_FIXED2},
+	{PNR_SERIES_RA4E2, 0x0100a100, 512, 0, 0, 0x08000000, 0x1000, 0x010080f0, 196, 0, 0, 0x20000000, 0xa000, 0, 0,
+		0x28000000, 0x400, RENESAS_FLASH_RV40, RENESAS_LOCATION_FIXED2},
+	{PNR_SERIES_RA4W1, 0x01010008, 44, 0, 0, 0x40100000, 0x2000, 0, 0, 0, 0, 0x20000000, 0x18000, 0, 0, 0, 0,
+		RENESAS_FLASH_MF3, RENESAS_LOCATION_FRT},
+	{PNR_SERIES_RA6M1, 0x01007000, 4096, 0x0100a150, 24, 0x40100000, 0x2000, 0, 0, 0x1ffe0000, 0x20000, 0x20000000,
+		0x20000, 0, 0, 0x200fe000, 0x2000, RENESAS_FLASH_RV40, RENESAS_LOCATION_FRT},
+	{PNR_SERIES_RA6M2, 0x01007000, 4096, 0x0100a150, 24, 0x40100000, 0x8000, 0, 0, 0x1ffe0000, 0x20000, 0x20000000,
+		0x40000, 0, 0, 0x200fe000, 0x2000, RENESAS_FLASH_RV40, RENESAS_LOCATION_FRT},
+	{PNR_SERIES_RA6M3, 0x01007000, 4096, 0x0100a150, 24, 0x40100000, 0x1000, 0, 0, 0x1ffe0000, 0x20000, 0x20000000,
+		0x40000, 0x20040000, 0x40000, 0x200fe000, 0x2000, RENESAS_FLASH_RV40, RENESAS_LOCATION_FRT},
+	{PNR_SERIES_RA6M4, 0x0100a100, 512, 0, 0, 0x08000000, 0x2000, 0x010080f0, 196, 0, 0, 0x20000000, 0x40000, 0, 0,
+		0x28000000, 0x400, RENESAS_FLASH_RV40, RENESAS_LOCATION_FIXED2},
+	{PNR_SERIES_RA6M5, 0x0100a100, 512, 0, 0, 0x08000000, 0x2000, 0x010080f0, 196, 0, 0, 0x20000000, 0x80000, 0, 0,
+		0x28000000, 0x400, RENESAS_FLASH_RV40, RENESAS_LOCATION_FIXED2},
+	{PNR_SERIES_RA6E1, 0x0100a100, 512, 0, 0, 0x08000000, 0x2000, 0x010080f0, 196, 0, 0, 0x20000000, 0x40000, 0, 0,
+		0x28000000, 0x400, RENESAS_FLASH_RV40, RENESAS_LOCATION_FIXED2},
+	{PNR_SERIES_RA6E2, 0x0100a100, 512, 0, 0, 0x08000000, 0x1000, 0x010080f0, 196, 0, 0, 0x20000000, 0xa000, 0, 0,
+		0x28000000, 0x400, RENESAS_FLASH_RV40, RENESAS_LOCATION_FIXED2},
+	{PNR_SERIES_RA6T1, 0x01007000, 4096, 0x0100a150, 24, 0x40100000, 0x2000, 0, 0, 0x1ffe0000, 0x10000, 0, 0, 0, 0, 0,
+		0, RENESAS_FLASH_RV40, RENESAS_LOCATION_FRT},
+	{PNR_SERIES_RA6T2, 0x0100a100, 512, 0, 0, 0x08000000, 0x4000, 0x010080f0, 196, 0, 0, 0x20000000, 0x10000, 0, 0,
+		0x28000000, 0x400, RENESAS_FLASH_RV40, RENESAS_LOCATION_FIXED2},
+};
+
+renesas_family_s renesas_family_lookup(renesas_pnr_series_e series)
+{
+	renesas_family_s details = {};
+	for (size_t i = 0; i < ARRAY_LENGTH(renesas_family); ++i)
+		if (renesas_family[i].series == series)
+			return renesas_family[i];
+	return details;
```

Comment:
It would be better for this to construct `details` on when it's actually needed and not on every call by doing:
```suggestion
	return (renesas_family_s){0};
```